### PR TITLE
Resolve Issue #142

### PR DESCRIPTION
**Pull request by james-d-mitchell**
I think this resolves Issue #142, I'm not sure how to test it exactly though, any input @isuruf?

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -216,6 +216,20 @@ AC_CHECK_FILE([$srcdir/extern/fmt-5.3.0/README.rst],
...
                [AC_MSG_ERROR([fmt-5.3.0 is not available, please download fmt-5.3.0 from https://fmt.dev/ and put the unpacked archive in the libsemigroups/extern directory])])
+
+# From https://stackoverflow.com/q/23379805/
+
+AC_CANONICAL_HOST
+X86_CXXFLAGS=""
+AS_CASE([$host_cpu],
+        [x86_64],
+        [
+         X86_CXXFLAGS="-mavx -march=native"
```
Comment (isuruf) [inline configure.ac]: Is it possible to add a configure option to disable this as well? For binary packages like conda packages `-march=native -mavx` are harmful as the built binaries can only be run on a newer architecture than the build machine.

Comment (james-d-mitchell) [reply configure.ac]: Sure thing, I'll add this to the PR now too

Comment (james-d-mitchell) [reply configure.ac]: Any name suggestions?

File: configure.ac [hunk 2]
```diff
@@ -216,6 +216,20 @@ AC_CHECK_FILE([$srcdir/extern/fmt-5.3.0/README.rst],
                [], 
                []
                [AC_MSG_ERROR([fmt-5.3.0 is not available, please download fmt-5.3.0 from https://fmt.dev/ and put the unpacked archive in the libsemigroups/extern directory])])
+
+# From https://stackoverflow.com/q/23379805/
+
+AC_CANONICAL_HOST
+X86_CXXFLAGS=""
+AS_CASE([$host_cpu],
+        [x86_64],
```
Comment (isuruf) [inline configure.ac]: On BSDs, this is `amd64`. See https://en.wikipedia.org/wiki/Uname

Comment (james-d-mitchell) [reply configure.ac]: A better solution will just be check if the flags are supported using `AX_CHECK_COMPILE_FLAG`.

File: configure.ac (hunk 1 above)
Comment (isuruf) [reply configure.ac]: NTL uses `--disable-native`. Btw, `-mavx -march=native` can be replaced by just `-march=native`

Comment (james-d-mitchell) [reply configure.ac]: Thanks, I'm working on this now.

Comment (james-d-mitchell) [reply configure.ac]: One more question @isuruf: should I account for the case where `-march=native` is not supported but `-mavx` is? If the answer is: no, then is the configure option `--disable-native` really required? The  setup I'm working on would not include the flag `-march=native` unless libsemigroups is compiled with `--enable-hpcombi`, so the flag (and HPCombi support) can be disabled by passing `--disable-hpcombi`. Thanks again

Comment (james-d-mitchell) [reply configure.ac]: Or maybe it'd be better to say: If `-march=native` is not supported , then with the changes in the current PR the flag won't be added.

**Comment by james-d-mitchell** [comment]
@isuruf Does this resolve the issue? If so, I'm happy to merge and make a release.

**Comment by isuruf** [comment]
Yes, it does. Thanks

**Pull request merged**